pkg/syncer/syncer.go

```diff
@@ -120,37 +159,57 @@ func (c *Controller) delete(ctx context.Context, gvr schema.GroupVersionResource
 }
 
 func (c *Controller) upsert(ctx context.Context, gvr schema.GroupVersionResource, namespace string, unstrob *unstructured.Unstructured) error {
+
 	client := c.getClient(gvr, namespace)
 
 	// Attempt to update the object; if the object isn't found, create it.
-	unstrob.SetResourceVersion("")
+
 	unstrob.SetUID("")
-	if _, err := client.Update(ctx, unstrob, metav1.UpdateOptions{}); err != nil {
-		if k8serrors.IsNotFound(err) {
-			if _, err := client.Create(ctx, unstrob, metav1.CreateOptions{}); err != nil {
-				return err
-			}
-		} else {
+	existing, err := client.Get(ctx, unstrob.GetName(), metav1.GetOptions{})
```

Comment:
This always-get-then-update-or-create results in us making 2x more API requests for each object change notification, vs just create-or-update, as before. Is there a reason this is more correct or robust?